spatio_temporal_voxel_layer/CMakeLists.txt

```diff
@@ -105,7 +105,6 @@ target_link_libraries(${library_name}
   ${Boost_LIBRARIES}
   ${EIGEN3_LIBRARIES}
   ${PCL_LIBRARIES}
-  ${OpenVDB_LIBRARIES}
```

Comment:
Why don't we link against OpenVDB? Does this happen automatically with the ament vendor package?

I checked the runtime libraries linked to `libspatio_temporal_voxel_layer_core.so` and OpenVDB was not there. Unless everything is done at compile time (which would mean we can skip building OpenVDB) something seems to be wrong here. I have NOT tried though, I'm just checking binaries and symbols so fat
```sh
	linux-vdso.so.1 (0x00007fff601c7000)
	libspatio_temporal_voxel_layer__rosidl_typesupport_cpp.so => /home/ivizzo/dev/spatio_temporal_voxel_layer/build/spatio_temporal_voxel_layer/libspatio_temporal_voxel_layer__rosidl_typesupport_cpp.so (0x000079e1e88fb000)
	libpcl_filters.so.1.12 => /lib/x86_64-linux-gnu/libpcl_filters.so.1.12 (0x000079e1e8236000)
	libpcl_common.so.1.12 => /lib/x86_64-linux-gnu/libpcl_common.so.1.12 (0x000079e1e819c000)
	libstd_srvs__rosidl_typesupport_cpp.so => /opt/ros/iron/lib/libstd_srvs__rosidl_typesupport_cpp.so (0x000079e1e8192000)
	liblaser_geometry.so => /opt/ros/iron/lib/liblaser_geometry.so (0x000079e1e8187000)
	libnav2_costmap_2d_core.so => /opt/ros/iron/lib/libnav2_costmap_2d_core.so (0x000079e1e8074000)
	libconsole_bridge.so.1.0 => /lib/x86_64-linux-gnu/libconsole_bridge.so.1.0 (0x000079e1e806e000)
	librcl.so => /opt/ros/iron/lib/librcl.so (0x000079e1e8027000)
	libtracetools.so => /opt/ros/iron/lib/libtracetools.so (0x000079e1e800c000)
	libclass_loader.so => /opt/ros/iron/lib/libclass_loader.so (0x000079e1e7ff8000)
	librclcpp_lifecycle.so => /opt/ros/iron/lib/librclcpp_lifecycle.so (0x000079e1e7fb2000)
	liborocos-kdl.so.1.5 => /lib/x86_64-linux-gnu/liborocos-kdl.so.1.5 (0x000079e1e7ef6000)
	libtf2.so => /opt/ros/iron/lib/libtf2.so (0x000079e1e7ed5000)
	libmessage_filters.so => /opt/ros/iron/lib/libmessage_filters.so (0x000079e1e7ed0000)
	librmw.so => /opt/ros/iron/lib/librmw.so (0x000079e1e7ec4000)
	librcutils.so => /opt/ros/iron/lib/librcutils.so (0x000079e1e7ea5000)
	librclcpp.so => /opt/ros/iron/lib/librclcpp.so (0x000079e1e7c5f000)
	libsensor_msgs__rosidl_typesupport_cpp.so => /opt/ros/iron/lib/libsensor_msgs__rosidl_typesupport_cpp.so (0x000079e1e7c4f000)
	liblibstatistics_collector.so => /opt/ros/iron/lib/liblibstatistics_collector.so (0x000079e1e7c48000)
	libstatistics_msgs__rosidl_typesupport_cpp.so => /opt/ros/iron/lib/libstatistics_msgs__rosidl_typesupport_cpp.so (0x000079e1e7c42000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x000079e1e7a16000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x000079e1e792f000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x000079e1e790d000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x000079e1e76e4000)
	/lib64/ld-linux-x86-64.so.2 (0x000079e1e8af5000)
	libspatio_temporal_voxel_layer__rosidl_generator_c.so => /home/ivizzo/dev/spatio_temporal_voxel_layer/build/spatio_temporal_voxel_layer/libspatio_temporal_voxel_layer__rosidl_generator_c.so (0x000079e1e76da000)
	librosidl_typesupport_cpp.so => /opt/ros/iron/lib/librosidl_typesupport_cpp.so (0x000079e1e76d4000)
	libpcl_sample_consensus.so.1.12 => /lib/x86_64-linux-gnu/libpcl_sample_consensus.so.1.12 (0x000079e1e6b06000)
	libpcl_search.so.1.12 => /lib/x86_64-linux-gnu/libpcl_search.so.1.12 (0x000079e1e691d000)
	libpcl_kdtree.so.1.12 => /lib/x86_64-linux-gnu/libpcl_kdtree.so.1.12 (0x000079e1e6807000)
	libpcl_octree.so.1.12 => /lib/x86_64-linux-gnu/libpcl_octree.so.1.12 (0x000079e1e65c1000)
	libgomp.so.1 => /lib/x86_64-linux-gnu/libgomp.so.1 (0x000079e1e6577000)
	libstd_srvs__rosidl_generator_c.so => /opt/ros/iron/lib/libstd_srvs__rosidl_generator_c.so (0x000079e1e6562000)
	libmap_msgs__rosidl_typesupport_cpp.so => /opt/ros/iron/lib/libmap_msgs__rosidl_typesupport_cpp.so (0x000079e1e654f000)
	libnav2_msgs__rosidl_typesupport_cpp.so => /opt/ros/iron/lib/libnav2_msgs__rosidl_typesupport_cpp.so (0x000079e1e64ec000)
	libnav2_util_core.so => /opt/ros/iron/lib/libnav2_util_core.so (0x000079e1e646a000)
	libnav_msgs__rosidl_typesupport_cpp.so => /opt/ros/iron/lib/libnav_msgs__rosidl_typesupport_cpp.so (0x000079e1e645c000)
	libtf2_ros.so => /opt/ros/iron/lib/libtf2_ros.so (0x000079e1e63d1000)
	libgeometry_msgs__rosidl_typesupport_cpp.so => /opt/ros/iron/lib/libgeometry_msgs__rosidl_typesupport_cpp.so (0x000079e1e63c4000)
	librcpputils.so => /opt/ros/iron/lib/librcpputils.so (0x000079e1e63b4000)
	libtinyxml2.so.9 => /lib/x86_64-linux-gnu/libtinyxml2.so.9 (0x000079e1e639c000)
	libament_index_cpp.so => /opt/ros/iron/lib/libament_index_cpp.so (0x000079e1e6391000)
	librcl_yaml_param_parser.so => /opt/ros/iron/lib/librcl_yaml_param_parser.so (0x000079e1e6385000)
	librmw_implementation.so => /opt/ros/iron/lib/librmw_implementation.so (0x000079e1e6378000)
	librcl_logging_spdlog.so => /opt/ros/iron/lib/librcl_logging_spdlog.so (0x000079e1e636d000)
	libyaml.so => /opt/ros/iron/lib/libyaml.so (0x000079e1e634b000)
	librcl_interfaces__rosidl_typesupport_c.so => /opt/ros/iron/lib/librcl_interfaces__rosidl_typesupport_c.so (0x000079e1e633b000)
	librcl_interfaces__rosidl_generator_c.so => /opt/ros/iron/lib/librcl_interfaces__rosidl_generator_c.so (0x000079e1e62ec000)
	librosidl_dynamic_typesupport.so => /opt/ros/iron/lib/librosidl_dynamic_typesupport.so (0x000079e1e62d2000)
	libtype_description_interfaces__rosidl_typesupport_c.so => /opt/ros/iron/lib/libtype_description_interfaces__rosidl_typesupport_c.so (0x000079e1e62ca000)
	libtype_description_interfaces__rosidl_generator_c.so => /opt/ros/iron/lib/libtype_description_interfaces__rosidl_generator_c.so (0x000079e1e62af000)
	librosidl_runtime_c.so => /opt/ros/iron/lib/librosidl_runtime_c.so (0x000079e1e628f000)
	liblttng-ust.so.1 => /lib/x86_64-linux-gnu/liblttng-ust.so.1 (0x000079e1e620c000)
	librcl_lifecycle.so => /opt/ros/iron/lib/librcl_lifecycle.so (0x000079e1e6202000)
	liblifecycle_msgs__rosidl_typesupport_cpp.so => /opt/ros/iron/lib/liblifecycle_msgs__rosidl_typesupport_cpp.so (0x000079e1e61f4000)
	liblifecycle_msgs__rosidl_typesupport_c.so => /opt/ros/iron/lib/liblifecycle_msgs__rosidl_typesupport_c.so (0x000079e1e61e8000)
	librcl_interfaces__rosidl_typesupport_cpp.so => /opt/ros/iron/lib/librcl_interfaces__rosidl_typesupport_cpp.so (0x000079e1e61cd000)
	librosgraph_msgs__rosidl_typesupport_cpp.so => /opt/ros/iron/lib/librosgraph_msgs__rosidl_typesupport_cpp.so (0x000079e1e61c8000)
	librcl_logging_interface.so => /opt/ros/iron/lib/librcl_logging_interface.so (0x000079e1e61c3000)
	libsensor_msgs__rosidl_generator_c.so => /opt/ros/iron/lib/libsensor_msgs__rosidl_generator_c.so (0x000079e1e617d000)
	libstatistics_msgs__rosidl_generator_c.so => /opt/ros/iron/lib/libstatistics_msgs__rosidl_generator_c.so (0x000079e1e6171000)
	libservice_msgs__rosidl_generator_c.so => /opt/ros/iron/lib/libservice_msgs__rosidl_generator_c.so (0x000079e1e616c000)
	libbuiltin_interfaces__rosidl_generator_c.so => /opt/ros/iron/lib/libbuiltin_interfaces__rosidl_generator_c.so (0x000079e1e6165000)
	liblz4.so.1 => /lib/x86_64-linux-gnu/liblz4.so.1 (0x000079e1e6145000)
	libmap_msgs__rosidl_generator_c.so => /opt/ros/iron/lib/libmap_msgs__rosidl_generator_c.so (0x000079e1e6111000)
	libnav2_msgs__rosidl_generator_c.so => /opt/ros/iron/lib/libnav2_msgs__rosidl_generator_c.so (0x000079e1e5f76000)
	libaction_msgs__rosidl_typesupport_cpp.so => /opt/ros/iron/lib/libaction_msgs__rosidl_typesupport_cpp.so (0x000079e1e5f6f000)
	libbondcpp.so => /opt/ros/iron/lib/libbondcpp.so (0x000079e1e5f00000)
	libnav_msgs__rosidl_generator_c.so => /opt/ros/iron/lib/libnav_msgs__rosidl_generator_c.so (0x000079e1e5ecf000)
	librclcpp_action.so => /opt/ros/iron/lib/librclcpp_action.so (0x000079e1e5eaf000)
	libtf2_msgs__rosidl_typesupport_cpp.so => /opt/ros/iron/lib/libtf2_msgs__rosidl_typesupport_cpp.so (0x000079e1e5ea0000)
	libgeometry_msgs__rosidl_generator_c.so => /opt/ros/iron/lib/libgeometry_msgs__rosidl_generator_c.so (0x000079e1e5e67000)
	libspdlog.so.1 => /lib/x86_64-linux-gnu/libspdlog.so.1 (0x000079e1e5dec000)
	librosidl_typesupport_c.so => /opt/ros/iron/lib/librosidl_typesupport_c.so (0x000079e1e5de6000)
	libnuma.so.1 => /lib/x86_64-linux-gnu/libnuma.so.1 (0x000079e1e5dd9000)
	liblttng-ust-common.so.1 => /lib/x86_64-linux-gnu/liblttng-ust-common.so.1 (0x000079e1e5dc9000)
	liblttng-ust-tracepoint.so.1 => /lib/x86_64-linux-gnu/liblttng-ust-tracepoint.so.1 (0x000079e1e5dab000)
	liblifecycle_msgs__rosidl_generator_c.so => /opt/ros/iron/lib/liblifecycle_msgs__rosidl_generator_c.so (0x000079e1e5d86000)
	librosgraph_msgs__rosidl_generator_c.so => /opt/ros/iron/lib/librosgraph_msgs__rosidl_generator_c.so (0x000079e1e5d81000)
	libstd_msgs__rosidl_generator_c.so => /opt/ros/iron/lib/libstd_msgs__rosidl_generator_c.so (0x000079e1e5d4f000)
	libgeographic_msgs__rosidl_generator_c.so => /opt/ros/iron/lib/libgeographic_msgs__rosidl_generator_c.so (0x000079e1e5d05000)
	libunique_identifier_msgs__rosidl_generator_c.so => /opt/ros/iron/lib/libunique_identifier_msgs__rosidl_generator_c.so (0x000079e1e5d00000)
	libaction_msgs__rosidl_generator_c.so => /opt/ros/iron/lib/libaction_msgs__rosidl_generator_c.so (0x000079e1e5cef000)
	libbond__rosidl_typesupport_cpp.so => /opt/ros/iron/lib/libbond__rosidl_typesupport_cpp.so (0x000079e1e5cea000)
	libuuid.so.1 => /lib/x86_64-linux-gnu/libuuid.so.1 (0x000079e1e5ce1000)
	librcl_action.so => /opt/ros/iron/lib/librcl_action.so (0x000079e1e5ccc000)
	libtf2_msgs__rosidl_generator_c.so => /opt/ros/iron/lib/libtf2_msgs__rosidl_generator_c.so (0x000079e1e5ca5000)
	libfmt.so.8 => /lib/x86_64-linux-gnu/libfmt.so.8 (0x000079e1e5c84000)
	libbond__rosidl_generator_c.so => /opt/ros/iron/lib/libbond__rosidl_generator_c.so (0x000079e1e5c7d000)
```

Reply:
It's implicit now. The package is added to the `dependencies` variable and link against later with: `ament_target_dependencies(${library_name} ${dependencies})`. As with all other ROS targets.

Reply:
I see, maybe my concerns are related to what @jmm-slamcore is seeing. Since on the library  shared object there are a bunch of undefined symbols to openvdb methods, and unless the lib plugin is loading the vendor package at runtime those won't be satisfied:
```sh
00000000001f0758 u guard variable for openvdb::v10_0::tree::LeafBuffer<openvdb::v10_0::tools::MeshToVoxelEdgeData::EdgeData, 3u>::sZero
00000000001f09f0 u guard variable for openvdb::v10_0::tree::LeafBuffer<double, 3u>::sZero
00000000001f0750 u guard variable for openvdb::v10_0::tree::LeafBuffer<int, 3u>::sZero
00000000001f0738 u guard variable for openvdb::v10_0::tree::Tree<openvdb::v10_0::tree::RootNode<openvdb::v10_0::tree::InternalNode<openvdb::v10_0::tree::InternalNode<openvdb::v10_0::tree::LeafNode<openvdb::v10_0::tools::MeshToVoxelEdgeData::EdgeData, 3u>, 4u>, 5u> > >::sTreeTypeName[abi:cxx11]
00000000001f09d8 u guard variable for openvdb::v10_0::tree::Tree<openvdb::v10_0::tree::RootNode<openvdb::v10_0::tree::InternalNode<openvdb::v10_0::tree::InternalNode<openvdb::v10_0::tree::LeafNode<double, 3u>, 4u>, 5u> > >::sTreeTypeName[abi:cxx11]
00000000001f0730 u guard variable for openvdb::v10_0::tree::Tree<openvdb::v10_0::tree::RootNode<openvdb::v10_0::tree::InternalNode<openvdb::v10_0::tree::InternalNode<openvdb::v10_0::tree::LeafNode<int, 3u>, 4u>, 5u> > >::sTreeTypeName[abi:cxx11]
00000000001f0760 u guard variable for openvdb::v10_0::points::TypedAttributeArray<unsigned char, openvdb::v10_0::points::GroupCodec>::sTypeName[abi:cxx11]
00000000001f0768 u guard variable for openvdb::v10_0::points::TypedAttributeArray<unsigned int, openvdb::v10_0::points::StringCodec<false> >::sTypeName[abi:cxx11]
00000000001f0830 u guard variable for openvdb::v10_0::math::Mat3<double>::identity()::sIdentity
00000000001f0888 u guard variable for openvdb::v10_0::math::Mat4<double>::identity()::sIdentity
00000000001f07a0 u openvdb::v10_0::tree::LeafBuffer<openvdb::v10_0::tools::MeshToVoxelEdgeData::EdgeData, 3u>::sZero
00000000001f09f8 u openvdb::v10_0::tree::LeafBuffer<double, 3u>::sZero
00000000001f0790 u openvdb::v10_0::tree::LeafBuffer<int, 3u>::sZero
00000000001f0740 u openvdb::v10_0::tree::Tree<openvdb::v10_0::tree::RootNode<openvdb::v10_0::tree::InternalNode<openvdb::v10_0::tree::InternalNode<openvdb::v10_0::tree::LeafNode<openvdb::v10_0::tools::MeshToVoxelEdgeData::EdgeData, 3u>, 4u>, 5u> > >::sTreeTypeName[abi:cxx11]
00000000001f09e8 u openvdb::v10_0::tree::Tree<openvdb::v10_0::tree::RootNode<openvdb::v10_0::tree::InternalNode<openvdb::v10_0::tree::InternalNode<openvdb::v10_0::tree::LeafNode<double, 3u>, 4u>, 5u> > >::sTreeTypeName[abi:cxx11]
00000000001f0748 u openvdb::v10_0::tree::Tree<openvdb::v10_0::tree::RootNode<openvdb::v10_0::tree::InternalNode<openvdb::v10_0::tree::InternalNode<openvdb::v10_0::tree::LeafNode<int, 3u>, 4u>, 5u> > >::sTreeTypeName[abi:cxx11]
00000000001f07b8 u openvdb::v10_0::points::TypedAttributeArray<unsigned char, openvdb::v10_0::points::GroupCodec>::sTypeName[abi:cxx11]
00000000001f07c0 u openvdb::v10_0::points::TypedAttributeArray<unsigned int, openvdb::v10_0::points::StringCodec<false> >::sTypeName[abi:cxx11]
00000000001f0840 u openvdb::v10_0::math::Mat3<double>::identity()::sIdentity
00000000001f08a0 u openvdb::v10_0::math::Mat4<double>::identity()::sIdentity
00000000001f09e0 u openvdb::v10_0::tree::Tree<openvdb::v10_0::tree::RootNode<openvdb::v10_0::tree::InternalNode<openvdb::v10_0::tree::InternalNode<openvdb::v10_0::tree::LeafNode<double, 3u>, 4u>, 5u> > >::treeType[abi:cxx11]()::once
00000000001bc9e0 u openvdb::v10_0::util::FindLowestOn(unsigned long)::DeBruijn
00000000001bc9c0 u openvdb::v10_0::util::FindHighestOn(unsigned int)::DeBruijn
```

Reply:
And sorry for the spam, but I can confirm the openvdb library is not being linked at all, This is the linker line:
<details><summary>Linker Line</summary>
<p>


```
[100%] Linking CXX shared library libspatio_temporal_voxel_layer_core.so
/usr/bin/cmake -E cmake_link_script CMakeFiles/spatio_temporal_voxel_layer_core.dir/link.txt --verbose=1
/usr/bin/c++ -fPIC  -std=c++17 -O2 -g -DNDEBUG -shared
-Wl,-soname,libspatio_temporal_voxel_layer_core.so -o libspatio_temporal_voxel_layer_core.so
CMakeFiles/spatio_temporal_voxel_layer_core.dir/src/spatio_temporal_voxel_layer.cpp.o
CMakeFiles/spatio_temporal_voxel_layer_core.dir/src/spatio_temporal_voxel_grid.cpp.o
CMakeFiles/spatio_temporal_voxel_layer_core.dir/src/measurement_buffer.cpp.o
CMakeFiles/spatio_temporal_voxel_layer_core.dir/src/frustum_models/depth_camera_frustum.cpp.o
CMakeFiles/spatio_temporal_voxel_layer_core.dir/src/frustum_models/three_dimensional_lidar_frustum.cpp.o
CMakeFiles/spatio_temporal_voxel_layer_core.dir/src/vdb2pc.cpp.o
-Wl,-rpath,/home/ivizzo/dev/spatio_temporal_voxel_layer/build/spatio_temporal_voxel_layer:/opt/ros/iron/lib:/home/ivizzo/underlay_ws/install/nav2_costmap_2d/lib:/home/ivizzo/underlay_ws/install/nav2_util/lib:/home/ivizzo/underlay_ws/install/nav2_msgs/lib:/home/ivizzo/underlay_ws/install/nav2_voxel_grid/lib
libspatio_temporal_voxel_layer__rosidl_typesupport_cpp.so
/usr/lib/x86_64-linux-gnu/libpcl_filters.so /usr/lib/x86_64-linux-gnu/libpcl_common.so
/usr/lib/x86_64-linux-gnu/libboost_system.so.1.74.0
/usr/lib/x86_64-linux-gnu/libboost_filesystem.so.1.74.0
/usr/lib/x86_64-linux-gnu/libboost_date_time.so.1.74.0
/usr/lib/x86_64-linux-gnu/libboost_iostreams.so.1.74.0
/usr/lib/x86_64-linux-gnu/libboost_serialization.so.1.74.0 /usr/lib/x86_64-linux-gnu/libflann_cpp.so
/opt/ros/iron/lib/libstd_srvs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libstd_srvs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libstd_srvs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libstd_srvs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libstd_srvs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libstd_srvs__rosidl_generator_py.so /opt/ros/iron/lib/liblaser_geometry.so
/opt/ros/iron/lib/libstatic_transform_broadcaster_node.so
/opt/ros/iron/lib/libvisualization_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libvisualization_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libvisualization_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libvisualization_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libvisualization_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libvisualization_msgs__rosidl_generator_py.so
/home/ivizzo/underlay_ws/install/nav2_costmap_2d/lib/liblayers.so
/home/ivizzo/underlay_ws/install/nav2_costmap_2d/lib/libfilters.so
/home/ivizzo/underlay_ws/install/nav2_costmap_2d/lib/libnav2_costmap_2d_core.so
/home/ivizzo/underlay_ws/install/nav2_costmap_2d/lib/libnav2_costmap_2d_client.so
/usr/lib/x86_64-linux-gnu/libconsole_bridge.so.1.0 /opt/ros/iron/lib/liblaser_geometry.so
/opt/ros/iron/lib/libmap_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libmap_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libmap_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libmap_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libmap_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libmap_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libmap_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libmap_msgs__rosidl_generator_py.so
/opt/ros/iron/lib/libmap_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libunique_identifier_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libunique_identifier_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libunique_identifier_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libunique_identifier_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libunique_identifier_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libunique_identifier_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libunique_identifier_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libunique_identifier_msgs__rosidl_generator_py.so
/opt/ros/iron/lib/libgeographic_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libgeographic_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libgeographic_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libgeographic_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libgeographic_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libgeographic_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libgeographic_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libgeographic_msgs__rosidl_generator_py.so
/home/ivizzo/underlay_ws/install/nav2_util/lib/libnav2_util_core.so
/home/ivizzo/underlay_ws/install/nav2_msgs/lib/libnav2_msgs__rosidl_generator_c.so
/home/ivizzo/underlay_ws/install/nav2_msgs/lib/libnav2_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libnav_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libgeographic_msgs__rosidl_typesupport_fastrtps_c.so
/home/ivizzo/underlay_ws/install/nav2_msgs/lib/libnav2_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libnav_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libgeographic_msgs__rosidl_typesupport_fastrtps_cpp.so
/home/ivizzo/underlay_ws/install/nav2_msgs/lib/libnav2_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libnav_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libgeographic_msgs__rosidl_typesupport_introspection_c.so
/home/ivizzo/underlay_ws/install/nav2_msgs/lib/libnav2_msgs__rosidl_typesupport_c.so
/home/ivizzo/underlay_ws/install/nav2_msgs/lib/libnav2_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libnav_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libgeographic_msgs__rosidl_typesupport_introspection_cpp.so
/home/ivizzo/underlay_ws/install/nav2_msgs/lib/libnav2_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libnav_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libgeographic_msgs__rosidl_typesupport_cpp.so
/home/ivizzo/underlay_ws/install/nav2_msgs/lib/libnav2_msgs__rosidl_generator_py.so
/home/ivizzo/underlay_ws/install/nav2_msgs/lib/libnav2_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libnav_msgs__rosidl_generator_py.so
/opt/ros/iron/lib/libgeographic_msgs__rosidl_generator_py.so
/opt/ros/iron/lib/libtf2_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libtf2_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libtf2_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libtf2_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libtf2_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libtf2_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libtf2_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libtf2_msgs__rosidl_generator_py.so /opt/ros/iron/lib/librclcpp_action.so
/opt/ros/iron/lib/liblifecycle_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/liblifecycle_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/liblifecycle_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/liblifecycle_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/liblifecycle_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/liblifecycle_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/liblifecycle_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/liblifecycle_msgs__rosidl_generator_py.so /opt/ros/iron/lib/librcl.so
/opt/ros/iron/lib/libtracetools.so -llttng-ust -llttng-ust-common -rdynamic -ldl
/opt/ros/iron/lib/librcl_lifecycle.so /opt/ros/iron/lib/libbondcpp.so
/opt/ros/iron/lib/libbond__rosidl_generator_c.so
/opt/ros/iron/lib/libbond__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libbond__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libbond__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libbond__rosidl_typesupport_c.so
/opt/ros/iron/lib/libbond__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libbond__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libbond__rosidl_generator_py.so /opt/ros/iron/lib/libbond__rosidl_typesupport_c.so
/opt/ros/iron/lib/libaction_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libaction_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libaction_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libaction_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libaction_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libaction_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libaction_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libaction_msgs__rosidl_generator_py.so
/opt/ros/iron/lib/librcl_interfaces__rosidl_generator_c.so
/opt/ros/iron/lib/librcl_interfaces__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/librcl_interfaces__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/librcl_interfaces__rosidl_typesupport_c.so
/opt/ros/iron/lib/librcl_interfaces__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/librcl_interfaces__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/librcl_interfaces__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/librcl_interfaces__rosidl_generator_py.so
/home/ivizzo/underlay_ws/install/nav2_voxel_grid/lib/libvoxel_grid.so
/opt/ros/iron/lib/libnav_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libnav_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libnav_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libnav_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libnav_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libnav_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libnav_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libnav_msgs__rosidl_generator_py.so
/opt/ros/iron/lib/libnav_msgs__rosidl_typesupport_c.so /opt/ros/iron/lib/libament_index_cpp.so
/opt/ros/iron/lib/libclass_loader.so /opt/ros/iron/lib/libclass_loader.so
/usr/lib/x86_64-linux-gnu/libtinyxml2.so /opt/ros/iron/lib/librclcpp_lifecycle.so
/opt/ros/iron/lib/librcl_lifecycle.so /opt/ros/iron/lib/libstd_srvs__rosidl_generator_c.so
/opt/ros/iron/lib/libstd_srvs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libstd_srvs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libstd_srvs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libstd_srvs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libstd_srvs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libstd_srvs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libstd_srvs__rosidl_generator_py.so
/opt/ros/iron/lib/libstd_srvs__rosidl_typesupport_c.so /usr/lib/x86_64-linux-gnu/liborocos-kdl.so
/opt/ros/iron/lib/libtf2.so /opt/ros/iron/lib/libtf2_ros.so
/opt/ros/iron/lib/libstatic_transform_broadcaster_node.so /opt/ros/iron/lib/libtf2_ros.so
/opt/ros/iron/lib/libbuiltin_interfaces__rosidl_generator_c.so
/opt/ros/iron/lib/libbuiltin_interfaces__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libbuiltin_interfaces__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libbuiltin_interfaces__rosidl_typesupport_c.so
/opt/ros/iron/lib/libbuiltin_interfaces__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libbuiltin_interfaces__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libbuiltin_interfaces__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libbuiltin_interfaces__rosidl_generator_py.so
/opt/ros/iron/lib/libgeometry_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libgeometry_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libgeometry_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libgeometry_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libgeometry_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libgeometry_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libgeometry_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libgeometry_msgs__rosidl_generator_py.so
/opt/ros/iron/lib/libvisualization_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libvisualization_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libvisualization_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libvisualization_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libvisualization_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libvisualization_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libvisualization_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libvisualization_msgs__rosidl_generator_py.so
/opt/ros/iron/lib/libvisualization_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libmessage_filters.so /opt/ros/iron/lib/libservice_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libservice_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libservice_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libservice_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libservice_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libservice_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libservice_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libservice_msgs__rosidl_generator_py.so
/opt/ros/iron/lib/librosidl_typesupport_fastrtps_c.so /opt/ros/iron/lib/librmw.so
/opt/ros/iron/lib/librosidl_typesupport_fastrtps_cpp.so /opt/ros/iron/lib/librcutils.so -ldl
/opt/ros/iron/lib/librcpputils.so /opt/ros/iron/lib/librosidl_typesupport_c.so
/opt/ros/iron/lib/librosidl_typesupport_cpp.so /opt/ros/iron/lib/librosidl_runtime_c.so
/opt/ros/iron/lib/librosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/librosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libpcl_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libpcl_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libsensor_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libpcl_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libsensor_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libpcl_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libpcl_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libsensor_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libpcl_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libsensor_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libpcl_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libpcl_msgs__rosidl_generator_py.so
/opt/ros/iron/lib/libpcl_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libsensor_msgs__rosidl_generator_py.so /opt/ros/iron/lib/librclcpp.so
/opt/ros/iron/lib/libsensor_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libsensor_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libsensor_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libsensor_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libsensor_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libsensor_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libsensor_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libsensor_msgs__rosidl_generator_py.so
/opt/ros/iron/lib/libsensor_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libsensor_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libstd_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libstd_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libstd_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libstd_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libstd_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libstd_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libstd_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libstd_msgs__rosidl_generator_py.so /usr/lib/x86_64-linux-gnu/libpython3.10.so
libspatio_temporal_voxel_layer__rosidl_generator_c.so
/usr/lib/x86_64-linux-gnu/libpcl_sample_consensus.so /usr/lib/x86_64-linux-gnu/libpcl_search.so
/usr/lib/x86_64-linux-gnu/libpcl_kdtree.so /usr/lib/x86_64-linux-gnu/libpcl_octree.so -llz4
/opt/ros/iron/lib/libstd_srvs__rosidl_generator_c.so /opt/ros/iron/lib/libmessage_filters.so
/opt/ros/iron/lib/librclcpp_action.so /opt/ros/iron/lib/librclcpp.so
/opt/ros/iron/lib/liblibstatistics_collector.so
/opt/ros/iron/lib/librosgraph_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/librosgraph_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/librosgraph_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/librosgraph_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/librosgraph_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/librosgraph_msgs__rosidl_generator_py.so
/opt/ros/iron/lib/librosgraph_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/librosgraph_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libstatistics_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libstatistics_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libstatistics_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libstatistics_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libstatistics_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libstatistics_msgs__rosidl_generator_py.so
/opt/ros/iron/lib/libstatistics_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libstatistics_msgs__rosidl_generator_c.so /opt/ros/iron/lib/librcl_action.so
/opt/ros/iron/lib/libtf2_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libaction_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libtf2_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libaction_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libtf2_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libaction_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libtf2_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libaction_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libtf2_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libaction_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libtf2_msgs__rosidl_generator_py.so
/opt/ros/iron/lib/libaction_msgs__rosidl_generator_py.so
/opt/ros/iron/lib/libtf2_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libtf2_msgs__rosidl_generator_c.so /opt/ros/iron/lib/libtf2.so
/usr/lib/x86_64-linux-gnu/libconsole_bridge.so.1.0
/opt/ros/iron/lib/libvisualization_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libmap_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libgeometry_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libstd_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libunique_identifier_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libgeometry_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libstd_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libunique_identifier_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libgeometry_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libstd_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libunique_identifier_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libgeometry_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libstd_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libunique_identifier_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libgeometry_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libstd_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libunique_identifier_msgs__rosidl_typesupport_cpp.so
/home/ivizzo/underlay_ws/install/nav2_msgs/lib/libnav2_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libaction_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libaction_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libnav_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libgeometry_msgs__rosidl_generator_py.so
/opt/ros/iron/lib/libstd_msgs__rosidl_generator_py.so
/opt/ros/iron/lib/libgeographic_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libgeographic_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libunique_identifier_msgs__rosidl_generator_py.so
/opt/ros/iron/lib/libunique_identifier_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libunique_identifier_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libbond__rosidl_generator_c.so
/opt/ros/iron/lib/liblifecycle_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/liblifecycle_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/liblifecycle_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/liblifecycle_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/liblifecycle_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/liblifecycle_msgs__rosidl_generator_py.so
/opt/ros/iron/lib/liblifecycle_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/liblifecycle_msgs__rosidl_generator_c.so /opt/ros/iron/lib/librcl.so
/opt/ros/iron/lib/librcl_logging_interface.so /opt/ros/iron/lib/librmw_implementation.so
/opt/ros/iron/lib/libament_index_cpp.so
/opt/ros/iron/lib/libtype_description_interfaces__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libtype_description_interfaces__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libtype_description_interfaces__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libtype_description_interfaces__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libtype_description_interfaces__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libtype_description_interfaces__rosidl_generator_py.so
/opt/ros/iron/lib/libtype_description_interfaces__rosidl_typesupport_c.so
/opt/ros/iron/lib/libtype_description_interfaces__rosidl_generator_c.so
/opt/ros/iron/lib/librcl_interfaces__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/librcl_interfaces__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/librcl_interfaces__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/librcl_interfaces__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/librcl_interfaces__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/librcl_interfaces__rosidl_generator_py.so
/opt/ros/iron/lib/librcl_interfaces__rosidl_typesupport_c.so
/opt/ros/iron/lib/librcl_interfaces__rosidl_generator_c.so
/opt/ros/iron/lib/librcl_yaml_param_parser.so
/opt/ros/iron/lib/libservice_msgs__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libbuiltin_interfaces__rosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/librosidl_typesupport_fastrtps_c.so
/opt/ros/iron/lib/libservice_msgs__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/libbuiltin_interfaces__rosidl_typesupport_fastrtps_cpp.so
/opt/ros/iron/lib/librosidl_typesupport_fastrtps_cpp.so /opt/ros/iron/lib/libfastcdr.so.1.0.27
/opt/ros/iron/lib/libservice_msgs__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libbuiltin_interfaces__rosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libservice_msgs__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/libbuiltin_interfaces__rosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/librosidl_typesupport_introspection_cpp.so
/opt/ros/iron/lib/librosidl_typesupport_introspection_c.so
/opt/ros/iron/lib/libservice_msgs__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/libbuiltin_interfaces__rosidl_typesupport_cpp.so
/opt/ros/iron/lib/librosidl_typesupport_cpp.so
/opt/ros/iron/lib/libservice_msgs__rosidl_generator_py.so
/opt/ros/iron/lib/libbuiltin_interfaces__rosidl_generator_py.so
/usr/lib/x86_64-linux-gnu/libpython3.10.so /opt/ros/iron/lib/librmw.so
/opt/ros/iron/lib/librosidl_dynamic_typesupport.so /opt/ros/iron/lib/libtracetools.so -llttng-ust
-llttng-ust-common -rdynamic -ldl /opt/ros/iron/lib/libgeometry_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libstd_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libpcl_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libsensor_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libgeometry_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libstd_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libservice_msgs__rosidl_typesupport_c.so
/opt/ros/iron/lib/libbuiltin_interfaces__rosidl_typesupport_c.so
/opt/ros/iron/lib/librosidl_typesupport_c.so /opt/ros/iron/lib/librcpputils.so
/opt/ros/iron/lib/libservice_msgs__rosidl_generator_c.so
/opt/ros/iron/lib/libbuiltin_interfaces__rosidl_generator_c.so
/opt/ros/iron/lib/librosidl_runtime_c.so /opt/ros/iron/lib/librcutils.so -ldl
gmake[2]: Leaving directory '/home/ivizzo/dev/spatio_temporal_voxel_layer/build/spatio_temporal_voxel_layer'
[100%] Built target spatio_temporal_voxel_layer_core
```

</p>
</details>

Reply:
@nachovizzo @Timple I added the plugin to the nav2 configuration in a tb3 simulation setup and I see an issue that relates to the discussion above:

`Could not load library dlopen error: nav2_tests/install/spatio_temporal_voxel_layer/lib/libspatio_temporal_voxel_layer_core.so: undefined symbol: _ZN7openvdb5v10_04math8internal4half5_eLutE, at ./src/shared_library.c:99`